src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -12,6 +12,17 @@ namespace Microsoft.Extensions.Configuration
     public interface IConfiguration
     {
         /// <summary>
+        /// Gets the key this element occupies in its parent or "" for the IConfigurationRoot.
+        /// </summary>
+        string Key { get; }
```

Comment:
yep, see https://github.com/aspnet/Configuration/pull/355 for a clean PR with the first stab at merging IConfigurationSection/IConfiguration